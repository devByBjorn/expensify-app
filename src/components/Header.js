import React from 'react'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'
import { startLogout } from '../actions/auth'


export const Header = ({ startLogout }) => (
  <div>
    <header>
      <h1>Expensify</h1>
      <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
      <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
      <button onClick={startLogout}>Logout</button>
    </header>
  </div>
)

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
})

export default connect(null, mapDispatchToProps)(Header)