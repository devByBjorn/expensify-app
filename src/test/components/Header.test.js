import React from 'react'
import { shallow } from 'enzyme'
// import toJson from 'enzyme-to-json'
import Header from '../../components/Header'

test('Should render Header corectly', () => {
  const wrapper = shallow(<Header />)
  expect(wrapper).toMatchSnapshot()
  // expect(toJson(wrapper)).toMatchSnapshot()
})
