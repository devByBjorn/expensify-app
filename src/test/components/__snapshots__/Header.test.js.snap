// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header corectly 1`] = `
<div>
  <header>
    <h1>
      Expensify
    </h1>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      to="/create"
    >
      Create Expense
    </NavLink>
  </header>
</div>
`;
